packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -21,6 +22,361 @@ import 'gesture_detector.dart';
+      _idToReferenceData[id] = referenceData;
+      group.add(referenceData);
+    } else {
+      referenceData = _idToReferenceData[id];
+      if (group.add(referenceData))
+        referenceData.count += 1;
+    }
+    return id;
+  }
+
+  /// Returns the Dart object associated with a reference id.
+  ///
+  /// The `groupName` parameter is not required by is added to regularize the
+  /// API surface of the methods in this class called from the Flutter IntelliJ
+  /// Plugin.
+  Object toObject(String id, [String groupName]) {
+    if (id == null)
+      return null;
+
+    final _InspectorReferenceData data = _idToReferenceData[id];
+    if (data == null)
+      throw new FlutterError('Id does not exist');
+    return data.object;
+  }
+
+  /// Returns the object to introspect to determine the source location of an
+  /// object's class.
+  ///
+  /// The Dart object for the id is returned for all cases but [Element] objects
+  /// where the [Widget] configuring the [Element] is returned instead as the
+  /// class of the [Widget] is more relevant than the class of the [Element].
+  ///
+  /// The `groupName` parameter is not required by is added to regularize the
+  /// API surface of methods called from the Flutter IntelliJ Plugin.
+  Object toObjectForSourceLocation(String id, [String groupName]) {
+    final Object object = toObject(id);
+    if (object is Element) {
+      return object.widget;
+    }
+    return object;
+  }
+
+  /// Remove the object with the specified `id` from the specified object
+  /// group.
+  ///
+  /// If the object exists in other groups it will remain alive and the object
+  /// id will remain valid.
+  void disposeId(String id, String groupName) {
+    if (id == null)
+      return;
+
+    final _InspectorReferenceData referenceData = _idToReferenceData[id];
+    if (referenceData == null)
+      throw new FlutterError('Id does not exist');
+    if (_groups[groupName]?.remove(referenceData) != true)
+      throw new FlutterError('Id is not in group');
+    _decrementReferenceCount(referenceData);
+  }
+
+  /// Set the [WidgetInspector] selection to the object matching the specified
+  /// id if the object is valid object to set as the inspector selection.
+  ///
+  /// Returns `true` if the selection was changed.
+  ///
+  /// The `groupName` parameter is not required by is added to regularize the
+  /// API surface of methods called from the Flutter IntelliJ Plugin.
+  bool maybeSetSelection(String id, [String groupName]) {
+    return maybeSetSelectionRaw(toObject(id), groupName);
+  }
+
+  /// Set the [WidgetInspector] selection to the specified `object` if it is
+  /// a valid object to set as the inspector selection.
+  ///
+  /// Returns `true` if the selection was changed.
+  ///
+  /// The `groupName` parameter is not needed but is specified to regularize the
+  /// API surface of methods called from the Flutter IntelliJ Plugin.
+  bool maybeSetSelectionRaw(Object object, [String groupName]) {
```

Comment:
do you intend this to be a public API?
If so, I would recommend a better pair of names than "maybeSetSelection" and "maybeSetSelectionRaw", though I can't quite tell what they do so it's hard to say what the names should be.

Reply:
yes this is intended to be public. the Flutter IntelliJ plugin sometimes wants to change selection given a raw Observatory object reference in which case the maybeSetSelectionRaw
api is used. Other times, the plugin wants to change selection given one of the WidgetInspectorService object ids in which case an id needs to be used.
I can remove the word maybe. It is there because the API ignores the request gracefully if the object doesn't happen to be an Element or RenderObject that the selection can be set to. I'm open to other name ideas for this method. I do agree these are two of the worse names I've come up with.

Reply:
how is the plugin calling these methods?

Reply:
I'd probably call them setSelectionById() and setSelection() respectively.

Reply:
done. Changed to setSelectionById and setSelection.
Here is the code in the plugin that is calling these methods. It is calling them by invoking code using the observatory protocol. It is a bit of a funky way of invoking them but it has the advantage that it is easy to use true Dart object references rather than just JSON as needed.
https://github.com/flutter/flutter-intellij/blob/master/src/io/flutter/inspector/InspectorService.java